### PR TITLE
Add TRT-RTX EP support, normalize alias, and force QDQ

### PR DESCRIPTION
**Pull request by anujj**
- Accept both "trt-rtx" and "NvTensorRtRtx" in --execution_provider, with
  runtime normalization to "trt-rtx" for internal handling.
- Automatically set extra_options["use_qdq"]=True when using TRT-RTX to
  ensure Q/DQ quantization path is enabled by default.

### CODE REVIEW

**Comment by anujj** [comment]
@gaugarg-nv @BLSharda @kunal-vaishnavi @baijumeswani

**Comment by anujj** [comment]
Updated PR to keep user facing name as NvTensorRtRtx

File: src/python/py/models/builder.py [hunk 1]
```diff
@@ -4634,5 +4638,7 @@ def get_args():
 
 if __name__ == '__main__':
     args = get_args()
+    if args.execution_provider == "NvTensorRtRtx":
```
Comment (kunal-vaishnavi) [inline src/python/py/models/builder.py]: We should handle this logic inside `create_model` so that users who import `create_model` can leverage this (e.g. Olive).

Comment (anujj) [reply src/python/py/models/builder.py]: i am forcing use_qdq in check_extra_options() with is called before create_model

    # force use_qdq for trt-rtx
    if args.execution_provider == "trt-rtx":
        kv_pairs["use_qdq"] = True
        
i had to switch the condition to `if args.execution_provider == "NvtensorRtRtx":`, which should be okey i guess ?

File: src/python/py/models/builder.py [hunk 2]
```diff
@@ -4325,6 +4325,10 @@ def check_extra_options(kv_pairs):
         # 'include_hidden_states' is for when 'hidden_states' are outputted and 'logits' are outputted
         raise ValueError("Both 'exclude_lm_head' and 'include_hidden_states' cannot be used together. Please use only one of them at once.")
 
+    # force use_qdq for trt-rtx
+    if args.execution_provider == "trt-rtx":
+        kv_pairs["use_qdq"] = True
```
Comment (kunal-vaishnavi) [inline src/python/py/models/builder.py]: In the original implementation of this before it was removed, there was a comment which said that QDQ was used because opset 21 was desired. The default opset is now 21 but it seems that creating a QDQ model is the true intention.

I also remember there was an issue in Olive which originated from forcing `use_qdq` here. Could there be a future scenario where creating a TRT-RTX model without QDQ is desired?

Comment (anujj) [reply src/python/py/models/builder.py]: we dont support matmulnbit in trt-rtx that's why it is always required, without this int4 path will break.

File: src/python/py/models/builder.py (hunk 1 above)
Comment (kunal-vaishnavi) [reply src/python/py/models/builder.py]: This is a separate issue from what you are mentioning. This check to change the EP name from `NvTensorRtRtx` to `trt-rtx` will not happen for users who import `create_model`. Here is an example which will fail with your current PR changes.

```python
from onnxruntime_genai.models.builder import create_model

input_dir = ""
output_dir = "./phi-4-mini"
cache_dir = "./cache_dir"

model_name = "microsoft/Phi-4-mini-instruct"
precision = "int4"
ep = "NvTensorRtRtx"

extra_options = {
    "use_qdq": "true"
}

create_model(model_name, input_dir, output_dir, precision, ep, cache_dir, **extra_options)
```

In the above code, `create_model` will not replace `NvTensorRtRtx` with `trt-rtx`. This will cause the incorrect ONNX model to get created. After moving the replacement of the EP name inside `create_model`, the above code should work.

File: src/python/py/models/builder.py (hunk 2 above)
Comment (kunal-vaishnavi) [reply src/python/py/models/builder.py]: Will `MatMulNBits` be supported in the future?

Comment (anujj) [reply src/python/py/models/builder.py]: No plan for that kunal

Comment (anujj) [reply src/python/py/models/builder.py]: simplify the changes in the latest commit

**Comment by gaugarg-nv** [comment]
@anujj we will also have to revert this PR in Olive: https://github.com/microsoft/Olive/pull/2169

**Comment by anujj** [comment]
Raised a PR for that : https://github.com/microsoft/Olive/pull/2182

File: src/python/py/models/builder.py [hunk 3]
```diff
@@ -4326,6 +4326,7 @@ def check_extra_options(kv_pairs):
         raise ValueError("Both 'exclude_lm_head' and 'include_hidden_states' cannot be used together. Please use only one of them at once.")
 
 
+
```
Comment (kunal-vaishnavi) [inline src/python/py/models/builder.py]:
```suggestion
```

File: src/python/py/models/builder.py [hunk 4]
```diff
@@ -4390,6 +4391,11 @@ def set_onnx_dtype(precision: str, extra_options: dict[str, Any]) -> ir.DataType
 
 @torch.no_grad
 def create_model(model_name, input_path, output_dir, precision, execution_provider, cache_dir, **extra_options):
+
```
Comment (kunal-vaishnavi) [inline src/python/py/models/builder.py]:
```suggestion
```
